### DISCUSSION TITLE
[Feature] Photo Request Feature - Ability to create a temporary link that allows anonymous users to upload media to an album/container

### DISCUSSION THREAD

**Discussion by rkkoszewski** [immich-app/immich#3941] [Feature Request]
### The feature

Example Scenario: You're travelling with friends and you want to share the pictures each person takes. You don't want to create a full blown account for every person, but instead create a temporary link that is shared with everyone and shows a media/file drop zone. This drop zone allows everybody to upload their images that then land in a shared album. This shared album can then be shared with all members of the group.

So the basic outline of the feature would be the following:
The ability to create a unique link with a expiration date that when accessed from a browser would just show a simple media upload interface without the need for the user to be logged in. Uploading media to this location will move the media to some container where these images can be managed and/or imported to the timeline or viewed somehow (An album?). 

Some things to consider:

- Anonymous user support? How to handle that? Security implications?
- What will the shared file location be. An Album? Directly to user feed (Impersonation?)? Something else?
- The link should be unique. It links to the current user and the current user can manage the link (Delete it, change expiry date, etc). Allow to set an optional expiry date and perhaps allow to specify a target drop zone for the files (Be it an album or other storage location, see previous point).
- It would be nice if the anonymous session could be remembered, so the user can manage the files that person dropped during the duration of that session (Delete unwanted pictures, etc).

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
If you create a shared album link, you can already enable uploading and set an expiry.

**Comment by MagikEh** [comment]
User @nasogaa has created something similar to what this request is looking for: https://github.com/Nasogaa/immich-drop

Note: The current method of 
> Make an album and enable user upload

Does not capture the desire to give out a single link at some event for patrons to upload their own photos on, preventing cross-user browsing, and hence why I stumbled across the above in my search for such functionality.
